master/getting-started/kubernetes/upgrade/downgrade.md

```diff
@@ -3,20 +3,41 @@ title: Downgrading Calico
 no_canonical: true
 ---
 
+## About downgrading {{site.prodname}}
+
 Under some circumstances, you may need to perform a downgrade and return your
-cluster to the previous version of {{site.prodname}}. You may need to do this
-before running `calico-upgrade complete` or afterwards. If you need to downgrade 
-your cluster after running `calico-upgrade complete`, you should do so as soon
-as possible to avoid an outage. Any pods created after `calico-upgrade complete`
-and before downgrading will lose networking.
+cluster to the previous version of {{site.prodname}}. If you need to downgrade
+you should do so as soon as possible to avoid an outage.
+
+The downgrade procedure varies according to how you originally installed
+{{site.prodname}} and your datastore type.
+
+- [Downgrading a self-hosted installation that uses the etcd
+  datastore](#downgrading-a-self-hosted-installation-that-uses-the-etcd-datastore)
+  is necessary when migration has been done and the {{site.prodname}}
+  components have been upgraded.
+  > **Important**: Any pods created after `calico-upgrade complete` and
+  > before downgrading will lose networking.
+  {: .alert .alert-danger}
+
+- [Aborting the upgrade when using the etcd
+  datastore](#aborting-the-upgrade-when-using-the-etcd-datastore) can be done
+  when the migration has been started but the {{site.prodname}} components
+  have not been upgraded.
 
-## Downgrading a self-hosted installation
+- [Downgrading a self-hosted installation that uses the Kubernetes API
+  datastore](#downgrading-a-self-hosted-installation-that-uses-the-kubernetes-api-datastore)
+
+- [Downgrading a custom installation](#downgrading-a-custom-installation)
+
+## Downgrading a self-hosted installation that uses the etcd datastore
 
 If you have upgraded {{site.prodname}} by deploying the latest manifest,
 follow the steps here to downgrade.
 
-1. Follow the steps in [Aborting the upgrade](#aborting-the-upgrade)
-   before downgrading the {{site.prodname}} components.
+1. Follow the steps in
+   [Aborting the upgrade](#aborting-the-upgrade-when-using-the-etcd-datastore)
+   can be done before downgrading the {{site.prodname}} components.
```

Comment:
During my test, I found reuse of `Aborting the upgrade` section could cause confusion because user is not actually `aborting upgrade`. Maybe something like ```Run `calico-upgrade abort` to instruct calico to use previous version datastore.```